app/models/factory.rb

```diff
@@ -70,6 +69,12 @@ def self.same_factory_overwrite(assos, same_factory)
     same_factory.reload
   end
 
+  # Return same name and parent factory trait from redis.
+  def self.same_trait(trait, factory_name)
+    traits = REDIS.smembers("traits").map{ |tr| JSON.parse(tr) }
+    traits.find { |tr| tr["name"] == trait && tr["factory_name"]== factory_name }
```

Comment:
Style/CollectionMethods: Prefer `detect` over `find`.<br>Style/SpaceAroundOperators: Surrounding space missing for operator '=='.